src/query/sql/src/planner/optimizer/hyper_dp/dphyp.rs

```diff
@@ -141,21 +141,56 @@ impl DPhpy {
                     };
                     self.filters.insert(filter);
                 }
-                let left_res = self.get_base_relations(
-                    &s_expr.children()[0],
-                    join_conditions,
-                    true,
-                    left_is_subquery,
-                    None,
-                )?;
-                let right_res = self.get_base_relations(
-                    &s_expr.children()[1],
-                    join_conditions,
-                    true,
-                    right_is_subquery,
-                    None,
-                )?;
-                Ok(left_res && right_res)
+                if left_is_subquery && right_is_subquery {
+                    // Parallel process subquery
+                    let metadata = self.metadata.clone();
+                    let left_expr = s_expr.children()[0].clone();
+                    let left_res = Thread::spawn(move || {
+                        let mut dphyp = DPhpy::new(metadata.clone());
+                        (dphyp.optimize(left_expr), dphyp.table_index_map)
+                    });
+                    let metadata = self.metadata.clone();
+                    let right_expr = s_expr.children()[1].clone();
+                    let right_res = Thread::spawn(move || {
+                        let mut dphyp = DPhpy::new(metadata.clone());
+                        (dphyp.optimize(right_expr), dphyp.table_index_map)
+                    });
```

Comment:
I'm not sure if it is necessary to parallelize the optimization here. And maybe it's not a good practice to spawn system threads here, @zhang2014 PTAL.

Reply:
+1, expect a best practice. Btw, it makes tpcds64 from 10s to 5s